typings/Controller.d.ts

```diff
@@ -1,22 +1,22 @@
 export type KeyState = 0 | 1 | 2;
 export type KeyTemplate = {
-  down: Function;
+  down?: () => void;
   key: string;
-  up: Function;
+  up?: () => void;
 }
 
 export class Key {
-  constructor(key: string, down: Function, up: Function);
+  constructor(key: string, down?: () => void, up?: () => void);
```

Comment:
The typing changes to the `Key` class does not match the actual [function signature](https://github.com/SpringRoll/SpringRoll/blob/develop/src/controller/Key.js#L17). The parameters `down` and `up` do not have default values and therefore will be set to `undefined` if none are provided, and not `noop` functions. I don't see anything wrong with this change, however I think the typings should reflect the actual code where possible. I think it should be fine if you wanted to also change the `Key` class to match this typings change.

Reply:
I think you misunderstand the syntax here. Declaration files are actually incapable of desceibing default parameters. The question mark here only says "this doesn't have to be provided" and the function type adds to rhat "but if it is, it has to be this". The current types requires you to pass in `null` or `undefined`. Being properly marked as optional will match the implementation.

Reply:
Ah! Yes, my mistake. In that case, I think we should be consistent with the other typing files and use `Function` to denote its type.

Reply:
I'll happily go through the rest of the files and replace all `Function` use if that is preferable. Currently about half the callbacks are already strongly typed in various files.

Reply:
Hi @andrewstart! I think that's great idea, given you have the time. I'll go ahead and get this merged shortly, and if you like you can make a PR to update those typings.